chainer/functions/array/get_item.py

```diff
@@ -15,55 +13,66 @@ class GetItem(function.Function):
     """Function that slices array and extract elements."""
 
     def __init__(self, slices):
-        if not isinstance(slices, collections.Iterable):
-            slices = tuple([slices])
+        if not isinstance(slices, tuple):
+            slices = slices,
 
         if chainer.is_debug():
             n_ellipses = 0
             for s in slices:
-                if numpy.isscalar(s) or s is None or isinstance(s, slice):
-                    pass
-                elif s is Ellipsis:
+                if s is Ellipsis:
                     n_ellipses += 1
-                else:
-                    raise ValueError('Only basic indexing is supported')
             if n_ellipses > 1:
                 raise ValueError('Only one Ellipsis is allowed')
 
         self.slices = slices
 
     def check_type_forward(self, in_types):
         type_check.expect(in_types.size() == 1)
-        valid_slice = len(self.slices) - self.slices.count(None)
+        n_nones = len([item for item in self.slices if item is None])
+        valid_slice = len(self.slices) - n_nones
         type_check.expect(in_types[0].ndim >= valid_slice)
 
     def forward(self, xs):
         ary = xs[0]
-        return utils.force_array(ary[tuple(self.slices)]),
+        return utils.force_array(ary[self.slices]),
 
     def backward(self, xs, gys):
         xp = cuda.get_array_module(*xs)
         gy = gys[0]
         gx = xp.zeros_like(xs[0])
-        gx[tuple(self.slices)] = gy
+        if xp is numpy:
+            numpy.add.at(gx, self.slices, gy)
+        else:
+            gx.scatter_add(self.slices, gy)
         return gx,
 
 
 def get_item(x, slices):
     """Extract elements from array with specified shape, axes and offsets.
 
     Args:
-        x (tuple of Variables): Variable to be sliced.
-        slices (int, slice, None or Ellipsis or tuple of them): Basic slicing
-            to slice a variable. It supports ``int``, ``slice``, ``newaxis``
-            (equivalent to ``None``) and ``Ellipsis``.
+        x (tuple of Variables): A variable to be sliced.
```

Comment:
Since this function accepts only one Variable, `(tuple of Variables)` should be `(~chainer.Variable)`, right?